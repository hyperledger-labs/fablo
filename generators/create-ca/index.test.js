const helpers = require('yeoman-test');
const path = require('path');
const fs = require('fs');

describe(':create-ca', () => {
  it('should create CA with defaults', async () => {
    const dir = await helpers.run(__dirname).toPromise();

    const file = fs.readFileSync(path.join(dir, '.yo-rc.json'));

    expect(file.toString()).toMatchSnapshot();
  });

  it('should create CA with custom options', async () => {
    const dir = await helpers.run(__dirname)
      .withOptions({ orgKey: 'black-pearl' })
      .withPrompts({ prefix: 'black-org-', generate: true })
      .toPromise();

    const file = fs.readFileSync(path.join(dir, '.yo-rc.json'));

    expect(file.toString()).toMatchSnapshot();
  });
});
