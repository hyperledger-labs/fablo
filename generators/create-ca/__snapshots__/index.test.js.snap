// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:create-ca should create CA 1`] = `
"{
  \\"generator-fabric-network\\": {
    \\"orgs\\": [
      {
        \\"ca\\": {
          \\"prefix\\": \\"black-org-\\",
          \\"generate\\": true
        }
      }
    ]
  }
}
"
`;

exports[`:create-ca should create CA with defaults 1`] = `
"{
  \\"generator-fabric-network\\": {
    \\"orgs\\": [
      {
        \\"ca\\": {
          \\"prefix\\": \\"ca\\",
          \\"generate\\": true
        }
      }
    ]
  }
}
"
`;
